### PR TITLE
[JBEE-249] Catch the NoClassDefFoundError and add to the notAClass co…

### PR DESCRIPTION
**Pull request by jamezp**
…llection for cases where the OS may have case sensitive paths.

### CODE REVIEW

**Comment by jamezp** [comment]
I tested on GlassFish 5.1 and I couldn't reproduce the issue so I'm not certain whether upstream needs this or not. It's possible it's just a JBoss Modules issue, but the stack trace on https://issues.jboss.org/browse/WFLY-12701 doesn't seem to indicate it would be a JBoss Modules issue.

File: api/src/main/java/javax/el/ImportHandler.java [hunk 1]
```diff
@@ -167,7 +167,11 @@ public ClassLoader run() {
                 }
 
                 return Class.forName(className, false, classLoader);
-            } catch (ClassNotFoundException ex) {
+                // Some operating systems have case sensitive path names. An example is Windows if className is
+                // attempting to be resolved from a wildcard import a java.lang.NoClassDefFoundError may be thrown as
+                // the case for the type likely doesn't match. See https://bugs.java.com/bugdatabase/view_bug.do?bug_id=8024775
+                // and https://bugs.openjdk.java.net/browse/JDK-8133522.
+            } catch (ClassNotFoundException | NoClassDefFoundError ex) {
```
Comment (scottmarlow) [inline api/src/main/java/javax/el/ImportHandler.java]: +1  

Are you going to create an [upstream pull request](https://github.com/eclipse-ee4j/el-ri) also?  Let me know if you prefer me to do that.

**Comment by jamezp** [comment]
@scottmarlow Looks like we each posted at the same time :) I can submit one, but I tested on GlassFish and I didn't seem to have the same issue. I'm not sure if they fix it in the implementation or how it's handled.

It looks like GlassFish ships the the one built from the [3.0.2-RI-COMB-RELEASE](https://github.com/eclipse-ee4j/el-ri/releases/tag/3.0.2-RI-COMB-RELEASE) tag so I'm just not sure what to do.

**Comment by scottmarlow** [comment]
I downloaded the source for 3.0.2-RI-COMB-RELEASE and it looks like they do catch ClassNotFoundException and handle in a similar way.  

I also looked for ClassNotFoundException use in jboss-modules and found one hit, 
[DefaultBootModuleLoaderHolder.java (not sure if this is the correct branch really)](https://github.com/jboss-modules/jboss-modules/commit/a6ee2a827bf59ab2906608f6db37b5260d0f6b41#diff-8023275fa8bb0102763de14497e6aee6R58) that catches ClassNotFoundException and throws NoClassDefFoundError instead.

**Comment by jamezp** [comment]
I'd have to take a better look, but I don't think that's the issue. I suppose I could however look deeper than just assuming this fix is needed :)

**Comment by scottmarlow** [comment]
I agree that the bug is not in jboss-modules, I think that el-api should really check for both exceptions, as your change accomplishes.  

The jboss-modules code that I found, seems to be changing a ClassNotFoundException into a NoClassDefFoundError, which might be why we see the NoClassDefFoundError.

**Comment by scottmarlow** [comment]
I'll merge this soon, if you don't create an upstream pr, I probably will.

**Pull request merged**